gravity-forms/gw-update-posts.php

```diff
@@ -140,27 +140,98 @@ public function update_post_by_entry( $entry, $form ) {
 		}
 
 		if ( $this->_args['terms'] ) {
-
 			// Assign custom taxonomies.
 			$term_fields = is_array( $this->_args['terms'] ) ? $this->_args['terms'] : array( $this->_args['terms'] );
+
 			foreach ( $term_fields as $field ) {
 				$term_field = GFAPI::get_field( $form, $field );
-				$terms      = array_map( 'intval', explode( ',', is_object( $term_field ) ? $term_field->get_value_export( $entry ) : '' ) );
-				$taxonomy   = is_object( $term_field ) ? $term_field['choices'][0]['object']->taxonomy : '';
 
-				wp_set_post_terms( $post->ID, $terms, $taxonomy );
+				$terms = explode( ',', is_object( $term_field ) ? trim( $term_field->get_value_export( $entry ) ) : '' );
+
+				if ( ! empty( $terms ) && rgar( $term_field, 'gppa-choices-enabled' ) ) {
+					$taxonomy = '';
+					// Get the taxonomy from the field settings or the first choice object based on the field type.
+					if ( $term_field instanceof GF_Field_Text ) {
+						$taxonomy = str_replace( 'taxonomy_', '', rgars( $term_field, 'gppa-values-templates/value', '' ) );
+					} elseif ( is_object( $term_field ) && ! empty( $term_field['choices'][0]['object']->taxonomy ) ) {
+						$taxonomy = $term_field['choices'][0]['object']->taxonomy;
+					} elseif ( is_object( $term_field ) && ! empty( $term_field['gppa-choices-filter-groups'][0][0]['property'] ) && $term_field['gppa-choices-filter-groups'][0][0]['property'] === 'taxonomy' ) {
```

Comment:
I like to leverage `rgars()` for stuff like this to make the conditional slightly shorter

```suggestion
					} elseif ( is_object( $term_field ) && rgars( $term_field, 'gppa-choices-filter-groups/0/0/property' ) === 'taxonomy' ) {
```

Reply:
(This one is an optional suggestion)

Reply:
Done!